### DISCUSSION TITLE
Set a prop when make a successful visit

### DISCUSSION THREAD

**Discussion by russellxrs** [inertiajs/inertia#1035] [Help] [answered]
What I want to do is to new a class based on one of the props when I receive them from backend, so I can use class with methods instead of plain objects. This is what I do:

I register a `success` event on the entry file:

```
Inertia.on('success', (event) => {
    event.detail.page.props.columnsMap = new ColumnsMap(event.detail.page.props.columnsMapOrigin);
});
```

When I click the inertia link, the `columnsMap` prop is successfully set while the progress bar is still moving.(I check it in vue dev tool)
But somehow it disapeared when the progress bar is ending. It's like beening reset when the visit finished.

So what is the problem?

**Comment by ajnsn** [comment, marked as answer]
Hey @russellxrs 
it is difficult to analyse why the described issue happens but in general you should never mutate a prop. Inertia had some support of transforming props and [removed it because of similar reasons](https://github.com/inertiajs/inertia/pull/693).

Instead you could use e.g. a computed property or a separate state to achieve the same behaviour.

Hope this helps :)
